File: gait/UFO_Waveform_Plotting.R
#Plotting script to plot individual participant waveforms and group means +/- sd
#Updated: 4/25/21


##A sad attempt at plotting multiple waveforms on a plot from Sunday 4/25/21
#Pre-Fatigue SS
setwd("/Users/rossbrancati/Desktop/UFO/Gait/Data/Participant_Mean_Waveforms/pre_fatigue_ss")
file_list <- list.files(path = "/Users/rossbrancati/Desktop/UFO/Gait/Data/Participant_Mean_Waveforms/pre_fatigue_ss")

pre_ss_all <- data.frame()

for (i in 1:length(file_list)){
    temp_file <- read.csv(file_list[i], header = FALSE)
    pre_ss_all <- rbind(pre_ss_all, temp_file)
}

ggplot(data=pre_ss_all) + 
  geom_line(aes(x=c(0:706),y=V4))


##Plotting Mean +/- SD 
pre_ss_walk = read.csv('/Users/rossbrancati/Desktop/UFO/Gait/Data/Ensemble_Waveforms/pre_fatigue_ss.csv', header=FALSE)
post_ss_walk = read.csv('/Users/rossbrancati/Desktop/UFO/Gait/Data/Ensemble_Waveforms/post_fatigue_ss.csv', header=FALSE)
pre_fast_walk = read.csv('/Users/rossbrancati/Desktop/UFO/Gait/Data/Ensemble_Waveforms/pre_fatigue_fast.csv', header=FALSE)
post_fast_walk = read.csv('/Users/rossbrancati/Desktop/UFO/Gait/Data/Ensemble_Waveforms/post_fatigue_fast.csv', header=FALSE)


current_data = post_fast_walk


ggplot() + 
  geom_ribbon(data=current_data, aes(x=c(0:100), ymin = (V5-V6), ymax = (V5+V6)), fill = rgb(136,28,28,maxColorValue=255), alpha=0.2) +
  geom_line(data=current_data, aes(x=c(0:100),y=V5), color=rgb(136,28,28,maxColorValue=255)) +
  scale_x_continuous(name="% Stance", expand = c(0,0), limits = c(1,100))+
  ylab ("Angle (Degrees)") + 
  ggtitle("Hip Flexion Angle - Stance") + 
  theme_classic() +
  theme(plot.title = element_text(hjust = 0.5))

ggplot() + 
  geom_ribbon(data=current_data, aes(x=c(0:100), ymin = (V7-V8), ymax = (V7+V8)), fill = rgb(136,28,28,maxColorValue=255), alpha=0.2) +
  geom_line(data=current_data, aes(x=c(0:100),y=V7), color=rgb(136,28,28,maxColorValue=255)) +
  scale_x_continuous(name="% Stance", expand = c(0,0), limits = c(1,100))+
  ylab ("Angle (Degrees)") + 
  ggtitle("Knee Flexion Angle - Stance") + 
  theme_classic() +
  theme(plot.title = element_text(hjust = 0.5))

ggplot() + 
  geom_ribbon(data=current_data, aes(x=c(0:100), ymin = (V13-V14), ymax = (V13+V14)), fill = rgb(136,28,28,maxColorValue=255), alpha=0.2) +
  geom_line(data=current_data, aes(x=c(0:100),y=V13), color=rgb(136,28,28,maxColorValue=255)) +
  scale_x_continuous(name="% Stance", expand = c(0,0), limits = c(1,100))+
  ylab ("Angle (Degrees)") + 
  ggtitle("Ankle Flexion Angle - Stance") + 
  theme_classic() +
  theme(plot.title = element_text(hjust = 0.5))

ggplot() + 
  geom_ribbon(data=current_data, aes(x=c(0:100), ymin = (V17-V18), ymax = (V17+V18)), fill = rgb(136,28,28,maxColorValue=255), alpha=0.2) +
  geom_line(data=current_data, aes(x=c(0:100),y=V17), color=rgb(136,28,28,maxColorValue=255)) +
  scale_x_continuous(name="% Stance", expand = c(0,0), limits = c(1,100))+
  ylab ("Angle (Degrees)") + 
  ggtitle("Hip Flexion Angle - Gait Cycle") + 
  theme_classic() +
  theme(plot.title = element_text(hjust = 0.5))

ggplot() + 
  geom_ribbon(data=current_data, aes(x=c(0:100), ymin = (V19-V20), ymax = (V19+V20)), fill = rgb(136,28,28,maxColorValue=255), alpha=0.2) +
  geom_line(data=current_data, aes(x=c(0:100),y=V19), color=rgb(136,28,28,maxColorValue=255)) +
  scale_x_continuous(name="% Stance", expand = c(0,0), limits = c(1,100))+
  ylab ("Angle (Degrees)") + 
  ggtitle("Knee Flexion Angle - Gait Cycle") + 
  theme_classic() +
  theme(plot.title = element_text(hjust = 0.5))

ggplot() + 
  geom_ribbon(data=current_data, aes(x=c(0:100), ymin = (V21-V22), ymax = (V21+V22)), fill = rgb(136,28,28,maxColorValue=255), alpha=0.2) +
  geom_line(data=current_data, aes(x=c(0:100),y=V21), color=rgb(136,28,28,maxColorValue=255)) +
  scale_x_continuous(name="% Stance", expand = c(0,0), limits = c(1,100))+
  ylab ("Angle (Degrees)") + 
  ggtitle("Ankle Flexion Angle - Gait Cycle") + 
  theme_classic() +
  theme(plot.title = element_text(hjust = 0.5))

ggplot() + 
  geom_ribbon(data=current_data, aes(x=c(0:100), ymin = (V31-V32), ymax = (V31+V32)), fill = rgb(136,28,28,maxColorValue=255), alpha=0.2) +
  geom_line(data=current_data, aes(x=c(0:100),y=V31), color=rgb(136,28,28,maxColorValue=255)) +
  scale_x_continuous(name="% Stance", expand = c(0,0), limits = c(1,100))+
  ylab ("Moment (Nm/kg)") + 
  ggtitle("Hip Flexion Moment") + 
  theme_classic() +
  theme(plot.title = element_text(hjust = 0.5))

ggplot() + 
  geom_ribbon(data=current_data, aes(x=c(0:100), ymin = (V34-V35), ymax = (V34+V35)), fill = rgb(136,28,28,maxColorValue=255), alpha=0.2) +
  geom_line(data=current_data, aes(x=c(0:100),y=V34), color=rgb(136,28,28,maxColorValue=255)) +
  scale_x_continuous(name="% Stance", expand = c(0,0), limits = c(1,100))+
  ylab ("Moment (Nm/kg)") + 
  ggtitle("Knee Flexion Moment") + 
  theme_classic() +
  theme(plot.title = element_text(hjust = 0.5))

ggplot() + 
  geom_ribbon(data=current_data, aes(x=c(0:100), ymin = (V39-V40), ymax = (V39+V40)), fill = rgb(136,28,28,maxColorValue=255), alpha=0.2) +
  geom_line(data=current_data, aes(x=c(0:100),y=V39), color=rgb(136,28,28,maxColorValue=255)) +
  scale_x_continuous(name="% Stance", expand = c(0,0), limits = c(1,100))+
  ylab ("Moment (Nm/kg)") + 
  ggtitle("Ankle Flexion Moment") + 
  theme_classic() +
  theme(plot.title = element_text(hjust = 0.5))

File: gait/UFO_Waveform_Plotting.m
%Plotting
%Individual waveforms will go on one plot
%Means +/- 

clear;
clc;

mean_dir = dir('/Users/rossbrancati/Desktop/UFO/Gait/Data/Participant_Mean_Waveforms');

i = 0;

participant_cell_array = cell(0,0);
participant_cell_array{1,1} = ['post_fatigue_fast'];
participant_cell_array{1,2} = ['post_fatigue_ss'];
participant_cell_array{1,3} = ['pre_fatigue_fast'];
participant_cell_array{1,4} = ['pre_fatigue_ss'];


for i = 4:length(mean_dir)
    
    current_folder = mean_dir(i,1);
    current_struct = dir(strcat(current_folder.folder,'/',current_folder.name));
    
    p = 0;
    sum_mat = zeros(101,24);
    
    for p = 3:length(current_struct)
        
        current_file_path = strcat(current_struct(p).folder,'/',current_struct(p).name);
        current_file = importdata(current_file_path);
        sum_mat = sum_mat + current_file;
        participant_cell_array{p-1,i-3} = current_file;
        
    end
end

vars = 0;
sub_id = 0;
cond = 0;
x = linspace(0,100,101)';

for cond = 1:4
    
    
    for sub_id = 2:length(participant_cell_array)
        
        figure(cond)
        %%HIP%%
        %Hip flexion angle stance
        hold on;
        subplot(4,4,1);
        plot(x,(participant_cell_array{sub_id,cond}(:,3)));
        title('Flexion Angle - Stance');
        ylabel('Hip','FontWeight','bold');

        %Hip flexion angle GC
        hold on;
        subplot(4,4,2);
        plot(x,(participant_cell_array{sub_id,cond}(:,9)));
        title('Flexion Angle - Gait Cycle');
        
        %Hip flexion moment
        hold on;
        subplot(4,4,3);
        plot(x,(participant_cell_array{sub_id,cond}(:,16)));
        title('Flexion Moment');
        
        %Hip Joint Power
        hold on;
        subplot(4,4,4);
        plot(x,(participant_cell_array{sub_id,cond}(:,21)));
        title('Joint Power');
        
        %%KNEE%%
        %Knee flexion angle stance
        hold on;
        subplot(4,4,5);
        plot(x,(participant_cell_array{sub_id,cond}(:,4)));
        ylabel('Knee','FontWeight','bold');
        
        %Knee flexion angle GC
        hold on;
        subplot(4,4,6);
        plot(x,(participant_cell_array{sub_id,cond}(:,10)));
        
        %Knee flexion moment
        hold on;
        subplot(4,4,7);
        plot(x,(participant_cell_array{sub_id,cond}(:,17)));
        
        %Knee Joint Power
        hold on;
        subplot(4,4,8);
        plot(x,(participant_cell_array{sub_id,cond}(:,22)));
        
        
        %%ANKLE%%
        %Ankle flexion angle stance
        hold on;
        subplot(4,4,9);
        plot(x,(participant_cell_array{sub_id,cond}(:,7)));
        ylabel('Ankle','FontWeight','bold');
        
        %Ankle flexion angle GC
        hold on;
        subplot(4,4,10);
        plot(x,(participant_cell_array{sub_id,cond}(:,11)));
        
        %Ankle flexion moment
        hold on;
        subplot(4,4,11);
        plot(x,(participant_cell_array{sub_id,cond}(:,20)));
        
        %Ankle Joint Power
        hold on;
        subplot(4,4,12);
        plot(x,(participant_cell_array{sub_id,cond}(:,23)));
        
        
        %%PELVIS%%
        %Pelvis Angle Stance (X)
        hold on;
        subplot(4,4,13);
        plot(x,(participant_cell_array{sub_id,cond}(:,12)));
        ylabel('Pelvis','FontWeight','bold');
        
        %Plevis Angle GC (X)
        hold on;
        subplot(4,4,14);
        plot(x,(participant_cell_array{sub_id,cond}(:,13)));        
        
    end
    
    legend('13','25','36','65','70','75','86');

end





